javascript/javascript3/week1/homework/weather-app/weather-app.js

```diff
@@ -0,0 +1,99 @@
+const loadIconElement = document.querySelector('.loadIcon');
+const weatherContainerElement = document.querySelector('.weatherInfoContainer');
+const weatherTempElement = document.querySelector('.weatherTemp');
+const weatherWindSpeedElement = document.querySelector('.weatherWindSpeed');
+const weatherCloudyPercentageElement = document.querySelector('.weatherCloudyPercentage');
+const weatherSunriseTimeElement = document.querySelector('.weatherSunriseTime');
+const weatherSunsetTimeElement = document.querySelector('.weatherSunsetTime');
+const weatherTypeIconElement = document.getElementById("weatherTypeIcon");
+const weatherCityNameElement = document.getElementById("weatherCityName");
+const weatherMapElement = document.querySelector('.weatherMap');
+const myLocationWeatherElement = document.getElementById("myLocationWeather");
+const locationStorageKey = "HYF.js3.w1.weather";
+
+buttonElement.addEventListener("click", () => {
+    const cityName = cityNameElement.value.toLowerCase().trim();
+    if (cityName == "") {
+        weatherContainerElement.style.visibility = "hidden";
+        inputMessageElement.innerText = "Please enter a city name inside the box!"
+        return;
+    } else {
+        loadIconElement.style.visibility = "visible";
+        inputMessageElement.innerHTML = "";
+        //A delay only to see the loading gif
+        setTimeout(function(){fetchWeather(cityName);}, 2000);
+    }
+})
+myLocationWeatherElement.addEventListener("click", getLocation);
+
+function showPosition(position) {
+    const latitude = position.coords.latitude;
+    const longitude = position.coords.longitude;
+    console.log(latitude);
+    console.log(longitude);
+
+    fetchMyWeather(latitude, longitude);
+}
+function getLocation() {
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(showPosition)
+    } else {
+        inputMessageElement.innerHTML = "Geolocation is not supported by this browser.";
+    }
+}
+function fetchMyWeather(lat, lon){
+    fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=f121c110a28ad82d5b470533a8560dc4`)
+    .then(response => response.json())
+    .then(myLocationWeather => renderWeather(myLocationWeather));
+}
+
+
+function fetchWeather(cityName){
+    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=f121c110a28ad82d5b470533a8560dc4`)
+    .then(response => response.json())
+    .then(cityWeather => renderWeather(cityWeather));
+}
+
+function renderWeather(weatherObj){
+    console.log(weatherObj);
+    weatherCityNameElement.innerHTML = weatherObj.name;
+    weatherTempElement.innerHTML = `${Math.floor(weatherObj.main.temp)}°`;
+    weatherWindSpeedElement.innerHTML = `${Math.floor(weatherObj.wind.speed)} m/s`;
+    weatherCloudyPercentageElement.innerHTML = `${Math.floor(weatherObj.clouds.all)} %`;
+    const sunriseTime = (new Date(weatherObj.sys.sunrise * 1000)).toLocaleTimeString();
```

Comment:
I think a function for this would be suitable. This has two benefits: 

1. It is easier to understand what is going on because if you choose a good function name, i dont have to know the details. 
2. If you all of a sudden want to change the time format, you can do it in one place